client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -17,6 +23,17 @@ export const requesting = createReducer( {}, {
 
 export const items = createReducer( {}, {
 	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+	[ ZONINATOR_UPDATE_ZONE ]: ( state, { siteId, data } ) => {
+		const zones = [ ...( state[ siteId ] || [] ) ];
+		const idx = findIndex( zones, { term_id: data.term_id || -1 } );
+
+		if ( idx < 0 ) {
+			return { ...state, [ siteId ]: zones.concat( [ data ] ) };
```

Comment:
Could we use `[ ...zones, data ]` instead? In this case, I find destructuring more intuitive. (Might be an obsolete point if you decide to follow [this strategy](https://github.com/Automattic/wp-calypso/pull/16953/files/c549e3b5bd40777dc606fa60c0bf4cba93b25c59#r132164990).)